### PR TITLE
Respect server's session tickets when possible even when using config…

### PR DESCRIPTION
**Pull request by oxtoacart**
…ured ClientSessionState

For getlantern/lantern-internal#3304

### CODE REVIEW

File: tlsdialer.go [hunk 1]
```diff
@@ -201,7 +201,14 @@ func (d *Dialer) DialForTimings(network, addr string) (*ConnWithTimings, error)
 	}
 	conn := tls.UClient(rawConn, configCopy, chid)
 	if d.ClientSessionState != nil {
-		conn.SetSessionState(d.ClientSessionState)
+		hasCachedSession := false
+		if configCopy.ClientSessionCache != nil {
+			sessionCacheKey := clientSessionCacheKey(result.ResolvedAddr, configCopy)
```
Comment (myleshorton) [inline tlsdialer.go]: Go uses `rawConn.GetRemoteAddr` for that first argument. Any reason not to do the same?

https://github.com/golang/go/blob/master/src/crypto/tls/handshake_client.go#L241

Comment (oxtoacart) [reply tlsdialer.go]: Sure, that should work. Will switch.

Comment (oxtoacart) [reply tlsdialer.go]: Okay, switched and verified that packet capture still looks good.

**Comment by myleshorton** [comment]
OK LGTM @oxtoacart!

**Comment by oxtoacart** [comment]
Thanks @myleshorton !

**Pull request merged**